src/borrow_tracker/tree_borrows/mod.rs

```diff
@@ -257,9 +249,10 @@ trait EvalContextPrivExt<'mir: 'ecx, 'tcx: 'mir, 'ecx>: crate::MiriInterpCxExt<'
         let Some(new_perm) = new_perm else { return Ok(Some((alloc_id, orig_tag))); };
 
         if let Some(protect) = new_perm.protector {
-            // See comment in `Stack::item_invalidated` for why we store the tag twice.
-            // FIXME: because of pointers that use parent tag, we need to handle items
-            // protected multiple times
+            // FIXME: because of pointers that use parent tag, we may need to handle items
+            // protected multiple times. For now this is fine since pointers that use the parent
+            // tag do not get a protector, but if we change the policy in
+            // `NewPermission::from_ref_ty` then we will need to also update it here.
```

Comment:
Currently the type of `new_perm` cannot even represent "old tag but please protect", so this comment is a bit confusing. I am not sure if it is actually necessary/helpful?